#  Web-scraping-bot project - simple data exploration

- Short exploration of a dataset about lifestyle and things to do in Tokyo from a specific website.
- The website's blog articles and related data have been directly scrapped from the web site.
- No data which could lead to this specific website will be shown in this notebook.

In [6]:
# import modules

import os
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

from scrappackage.scrap_blog_articles import Website_blog

In [2]:
# magic commands

%load_ext autoreload
%autoreload 2
%matplotlib inline

In [3]:
# The whole dataset have been scrapped before the creation of this notebook, a week ago.
# Since, some articles could have been published.
# Update the csv file which contains the dataset:
# To avoid scrap all the website's blog articles, update
# the all_articles.csv file with the website's blog articles from the
# first page only. This update could be done every two week as there were
# no more than 25 article publications per month in the past.

# instantiate the Website_blog class.
wb = Website_blog()

# update the csv file:
wb.update_csv_file_with_blog_first_page()

Scrap articles from the website's page 1
****************** 18 articles scrapped on this page.

'all_articles.csv' is now up to date.


In [8]:
# all_articles.csv relative path
all_articles_csv_file_path = os.path.join("..", "scrappackage", "all_articles.csv")
                                                                
df = pd.read_csv(all_articles_csv_file_path)

In [15]:
df.columns

Index(['url', 'title', 'date', 'author', 'categories', 'tags', 'text'], dtype='object')

In [16]:
df.shape

(1624, 7)

The dataframe contains:   
    - 1624 articles   
    - 7 columns: url, title, date, author, categories, tags, text

In [21]:
df.count()

url           1624
title         1624
date          1535
author        1535
categories    1624
tags          1501
text          1624
dtype: int64

In [22]:
df.nunique()

url           1606
title         1605
date          1268
author         104
categories     176
tags          1385
text          1594
dtype: int64

In [34]:
# to be continued